core/src/main/java/tc/oc/pgm/map/source/GitMapSourceFactory.java

```diff
@@ -0,0 +1,152 @@
+package tc.oc.pgm.map.source;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Stream;
+import org.eclipse.jgit.api.CloneCommand;
+import org.eclipse.jgit.api.FetchCommand;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.ResetCommand;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.api.errors.TransportException;
+import org.eclipse.jgit.transport.CredentialsProvider;
+import org.eclipse.jgit.transport.UsernamePasswordCredentialsProvider;
+import tc.oc.pgm.api.map.MapSource;
+import tc.oc.pgm.api.map.exception.MapMissingException;
+
+public class GitMapSourceFactory extends SystemMapSourceFactory {
+
+  private final String gitURI;
+  private final Logger logger;
+  private final File dir;
+  // The factory needs to parse the source differently depending on if credentials are provided
+  private static boolean credentialsProvided;
+  private boolean updateOnRestart;
+
+  private CredentialsProvider provider = null;
+
+  private static File getFile(String source) {
+    credentialsProvided = isCredentialsProvided(source);
+
+    int uniqueIdentifier =
+        Math.abs(source.hashCode()); // Prevents errors on similar repository names
+
+    String dirName;
+    if (credentialsProvided)
+      dirName =
+          (source.substring(source.lastIndexOf("/"), source.indexOf(";")) + "_" + uniqueIdentifier);
+    else dirName = (source.substring(source.lastIndexOf("/") + 1) + "_" + uniqueIdentifier);
+
+    return new File("./gitMaps/" + dirName);
+  }
+
+  public GitMapSourceFactory(String source, Logger logger, boolean updateOnRestart)
+      throws MapMissingException, ClassNotFoundException {
+    super(getFile(source).getPath());
+
+    if (credentialsProvided) this.gitURI = source.substring(0, source.indexOf(";"));
+    else this.gitURI = source;
+
+    this.logger = logger;
+    this.dir = getFile(source);
+    this.updateOnRestart = updateOnRestart;
+
+    // Disable if JGit is not present
+    if (Class.forName("org.eclipse.jgit.api.Git") == null) {
```

Comment:
You need to do this check before calling the `GitMapSourceFactory` constructor, or else it won't work.